File: lib/player.go
package lib

import "fmt"

// Player struct
type Player struct {
	Name   string
	Game   *Game
	Pocket Cards
}

// BestHand ...
func (player *Player) BestHand() Hand {
	return Hands(player.Pocket, player.Game.Board)[0]
}

// String ...
func (player *Player) String() string {
	return fmt.Sprintf(
		"name:%s\tpocket:%s\t%s\t%s",
		player.Name,
		player.Pocket.String(),
		player.BestHand().String(),
		BestPossible(player.Pocket, player.Game.Board, Cards(player.Game.Deck))[0].String())
}

// Players type
type Players []*Player

File: lib/Player.js
import Hand from './Hand'
import Nuts from './Nuts'

export default class Player {
  constructor(id, name) {
    this.id = id
    this.name = name || id
    this.cards = []
  }
  async init(game) {
    await this.sleep()
    this.game = game
    this.cards.length = 0
  }
  get hand() {
    return new Hand(this.cards, this.game ? this.game.board : [])
  }
  get nuts() {
    return new Nuts(this.cards, this.game ? this.game.board : [], this.game ? this.game.deck : [])
  }
  sleep(time) {
    return new Promise(resolve => setTimeout(resolve, time || 10))
  }
}
